src/context.ts

```diff
@@ -865,6 +865,21 @@ export class Context<U extends Deunionize<tg.Update> = tg.Update> {
     })
   }
 
+  persistentChatAction(action: Shorthand<'sendChatAction'>[0], callback: () => Promise<void>, extra?: tt.ExtraSendChatAction) {
+    return new Promise<void>(async (resolve) => {
+      await this.sendChatAction(action, {...extra})
+
+      const timer = setInterval(
```

Comment:
Using a `setInterval` loop makes any errors in `sendChatAction` unhandle-able. This is bad because the process will crash if it errors.

But also we may want such errors to be ignored, because what we really care about is the actual long process completing and performing an action. I do not think we'd ever want a failure in `sendChatAction` to abort the handler early. So the best course of action is to do:

```TS
setInterval(
	async () => this
		.sendChatAction(action, { ...extra })
		.catch(err => { debug("Ignored error while persisting sendChatAction:", err) })
	...
```

You'll need to instantiate debug as `d("telegraf:context")` in this file, similar to

https://github.com/telegraf/telegraf/blob/5137c75b249898cfdf35bb1560d649f6a7658e62/src/telegraf.ts#L19

This error-ignoring behaviour must be documented in the method's JSDoc, along with a description of what the function itself.